pkg/logging/formatter.go

```diff
@@ -0,0 +1,164 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"golang.org/x/exp/slog"
+)
+
+type formatter struct {
+	l *slog.Logger
+
+	exiter   exiter
+	panicker panicker
+}
+
+type exiter interface {
+	Exit(code int)
+}
+
+type defaultExiter struct{}
+
+func (de defaultExiter) Exit(code int) {
+	os.Exit(code)
+}
+
+type panicker interface {
```

Comment:
Are there multiple ways to panic? :rofl:

Reply:
Once you have kids, there are countably infinite ways to panic 😄

Reply:
:see_no_evil: